OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitMobileNotSupportedFragment.kt

```diff
@@ -0,0 +1,180 @@
+
+        binding.toggleShow.setOnClickListener {
+            val showMore = binding.layoutUpgradeFeature.containerLayout.isInvisible()
+            binding.layoutUpgradeFeature.containerLayout.visibility(showMore)
+            binding.toggleShow.text = getText(
+                if (showMore) R.string.course_modal_graded_assignment_show_less
+                else R.string.course_modal_graded_assignment_show_more
+            )
+            unit?.let {
+                environment.analyticsRegistry.trackValuePropShowMoreLessClicked(
+                    it.courseId, it.id, price, isSelfPaced, showMore
+                )
+            }
+        }
+    }
+
+    private fun showNotAvailableOnMobile() {
+        binding.containerLayoutNotAvailable.visibility = View.VISIBLE
+        binding.llGradedContentLayout.visibility = View.GONE
+        binding.contentErrorIcon.setImageDrawable(getDrawable(requireContext(), R.drawable.ic_lock))
+        binding.notAvailableMessage.setText(R.string.not_available_on_mobile)
+        binding.notAvailableMessage2.visibility = View.GONE
+    }
+
+    private fun showOnlyAvailableOnWeb() {
+        binding.containerLayoutNotAvailable.visibility = View.VISIBLE
+        binding.llGradedContentLayout.visibility = View.GONE
+        binding.contentErrorIcon.setImageDrawable(
+            getDrawable(requireContext(), R.drawable.ic_laptop)
+        )
+        binding.notAvailableMessage.setText(
+            if (unit?.type === BlockType.VIDEO) R.string.video_only_on_web_short
+            else R.string.assessment_not_available
+        )
+        binding.notAvailableMessage2.visibility = View.VISIBLE
+    }
+
+    private fun setUpUpgradeButton(isSelfPaced: Boolean, price: String) {
+        if (environment.config.isIAPEnabled) {
+            binding.layoutUpgradeBtn.root.visibility = View.VISIBLE
+            binding.layoutUpgradeBtn.btnUpgrade.setOnClickListener {
+                enableUpgradeButton(false)
+                purchaseProduct("org.edx.mobile.test_product")
+                unit?.let {
+                    environment.analyticsRegistry.trackUpgradeNowClicked(
+                        it.courseId, price, it.id, isSelfPaced
+                    )
+                }
+            }
+
+            billingProcessor = BillingProcessor(requireContext(), object : BillingFlowListeners {
+                override fun onPurchaseCancel() {
+                    enableUpgradeButton(true)
+                }
+
+                override fun onPurchaseComplete(purchase: Purchase) {
+                    CoroutineScope(Dispatchers.Main).launch {
+                        enableUpgradeButton(true)
+                    }
+                    AlertDialogFragment.newInstance(
+                        getString(R.string.title_upgrade_complete),
+                        getString(R.string.upgrade_success_message),
+                        getString(R.string.label_continue).toUpperCase(Locale.ROOT),
```

Comment:
no need to add UpperCase property as AlertDialog action by default are in the upper case.